### DISCUSSION TITLE
How to take necking into uniaxial tension simulation

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#19325] [Q&A Modules: Solid mechanics] [answered]
Dear Moose experts or users,

Could anyone share any ideas or thoughts about how to take the necking mechanism/phenomenon into the power-law model for uniaxial tension simulation ( https://mooseframework.inl.gov/source/materials/crystal_plasticity/CrystalPlasticityKalidindiUpdate.html) with me? So that I can observe when the fracture will happen on the stress-strain curve. Thank you for your help.

**Comment by hugary1995** [comment, marked as answer]
Necking is typically due to inelastic mechanisms. Plasticity is one of the mechanisms that will lead to necking. It is not specific to any hardening law. It depends on the geometry and material properties.

Fracture is a different phenomenon that isn't necessarily related to necking. A ductile specimen may fail before or after necking is observed.
